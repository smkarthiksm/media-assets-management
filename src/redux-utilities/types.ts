import { Action, ThunkAction } from '@reduxjs/toolkit';
import { reducers, store } from './store';

export type RootState = ReturnType<typeof reducers>;
export type AppDispatch = typeof store.dispatch;

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action
>;
