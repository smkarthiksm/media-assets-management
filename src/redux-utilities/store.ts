import { combineReducers, configureStore } from '@reduxjs/toolkit';
import loginReducer from './slices/auth-slice';

export const reducers = combineReducers({
  login: loginReducer,
});

export const store = configureStore({
  reducer: reducers,
});
