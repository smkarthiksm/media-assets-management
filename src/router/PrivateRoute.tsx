import { Outlet, Navigate } from 'react-router-dom';
import { isAuthenticated } from '../components/utilities/utility';

const PrivateRoute = () => {
  return isAuthenticated() ? <Outlet /> : <Navigate to="/login" replace />;
};

export default PrivateRoute;
