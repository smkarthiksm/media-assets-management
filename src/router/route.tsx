import { Navigate, RouteObject, createBrowserRouter } from 'react-router-dom';
import LoginComponent from '../components/login/login';
import HomeComponent from '../components/home/home';
import PrivateRoute from './PrivateRoute';
import { isAuthenticated } from '../utilities/utility';
import AllFilesSectionComponent from '../components/all-files-section/all-files-section';

export const routesConfig: RouteObject[] = [
  {
    path: '/',
    element: (
      <Navigate to={`/${isAuthenticated() ? 'home' : 'login'}`} replace />
    ),
  },
  {
    element: <PrivateRoute />,
    children: [
      {
        path: '/home',
        element: <HomeComponent />,
        children: [
          {
            path: '',
            element: <AllFilesSectionComponent />,
          },
        ],
      },
    ],
  },
  {
    path: '/login',
    element: !isAuthenticated() ? (
      <LoginComponent />
    ) : (
      <Navigate to="/home" replace />
    ),
  },
  {
    path: '*',
    element: <Navigate to="/" replace />,
  },
];

export const router = createBrowserRouter(routesConfig);
