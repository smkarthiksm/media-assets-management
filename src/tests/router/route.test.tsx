import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import { routesConfig } from '../../router/route';
import { Provider } from 'react-redux';
import { appStore } from '../../redux-utilities/store';

const renderWithProvider = () => {
  const router = createMemoryRouter(routesConfig, {
    initialEntries: ['/'],
  });
  return render(
    <Provider store={appStore({})}>
      <RouterProvider router={router} />
    </Provider>,
  );
};
describe('Router component', () => {
  describe('Login Route', () => {
    test('should load login page', () => {
      const { container } = renderWithProvider();

      expect(container).toMatchSnapshot();
      expect(screen.getByRole('button', { name: 'Login' })).toBeInTheDocument();
    });

    test('should load login page when not logged in', () => {
      renderWithProvider();

      expect(screen.getByRole('button', { name: 'Login' })).toBeInTheDocument();
    });

    test('should load login page when not logged in and requested bad route', () => {
      renderWithProvider();

      expect(screen.getByRole('button', { name: 'Login' })).toBeInTheDocument();
    });
  });
});
