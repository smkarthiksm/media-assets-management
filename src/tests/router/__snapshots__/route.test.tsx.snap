// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router component Login Route should load login page 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 login-container css-1i4yeyn-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column logo-container css-1emcw5a-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <span
          class="logo"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        >
          Media Assets Management
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item seperator-container css-13i4rnv-MuiGrid-root"
      >
        <div
          class="seperator mt-2"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root input-field mt-4 css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r0:"
              name="email"
              placeholder="Email"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root input-field mt-3 css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r1:"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth mt-3 css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          Login
        </button>
      </div>
    </div>
  </div>
</div>
`;
